internal/pkg/overlay/overlay.go

```diff
@@ -668,19 +773,9 @@ func CarefulWriteBuffer(destFile string, buffer bytes.Buffer, backupFile bool, p
 	return err
 }
 
-/*
-Parses the template with the given filename, variables must be in data. Returns the
-parsed template as bytes.Buffer, and the bool variables for backupFile and writeFile.
-If something goes wrong an error is returned.
-*/
-func RenderTemplateFile(fileName string, data TemplateStruct) (
-	buffer bytes.Buffer,
-	backupFile bool,
-	writeFile bool,
-	err error,
-) {
-	backupFile = true
-	writeFile = true
+// getTemplateFuncMap returns a template.FuncMap with all the functions
+// for warewulf templates.
+func getTemplateFuncMap(fileName string, data TemplateStruct, writeFile *bool, backupFile *bool) template.FuncMap {
```

Comment:
getTemplateFuncMap should allocate writeFile and backupFile internally, and return those pointers along with the FuncMap.